### ISSUE TITLE
[FEA] support for groupby named aggregates

### ISSUE THREAD

**Issue by mattf** [rapidsai/cudf#12524]
**Is your feature request related to a problem? Please describe.**
writing code with `import cudf as pd`

**Describe the solution you'd like**
`groupby().agg()` matching `pandas`

https://pandas.pydata.org/docs/user_guide/groupby.html#groupby-aggregate-named

e.g.
```
In [1]: import cudf as pd

In [2]: pd.__version__
Out[2]: '22.12.0'

In [3]: df = pd.DataFrame({"size": ["S", "S", "M", "L"], "price": [44, 29.99, 10, 19]})

In [4]: df.groupby('size').agg({'price': 'mean'})
Out[4]: 
       price
size        
L     19.000
M     10.000
S     36.995

In [5]: df.groupby('size').agg(mean_price=('price', 'mean'))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [5], line 1
----> 1 df.groupby('size').agg(mean_price=('price', 'mean'))

File ~/.local/lib/python3.9/site-packages/nvtx/nvtx.py:101, in annotate.__call__.<locals>.inner(*args, **kwargs)
     98 @wraps(func)
     99 def inner(*args, **kwargs):
    100     libnvtx_push_range(self.attributes, self.domain.handle)
--> 101     result = func(*args, **kwargs)
    102     libnvtx_pop_range(self.domain.handle)
    103     return result

TypeError: agg() got an unexpected keyword argument 'mean_price'

In [6]: df.to_pandas().groupby('size').agg(mean_price=('price', 'mean'))
Out[6]: 
      mean_price
size            
L         19.000
M         10.000
S         36.995

```

**Comment by Matt711** [comment]
Closed by #16528

**Comment by Matt711** [comment]
Closed by #16528
